generator/src/main/java/org/corfudb/generator/State.java

```diff
@@ -111,7 +112,7 @@ public long stopTx() {
         return runtime.getObjectsView().TXEnd();
     }
 
-    public void startSnapshotTx(long snapshot) {
+    public void startSnapshotTx(Token snapshot) {
```

Comment:
Should lastSuccessfulWriteOperationTimestamp and lastSuccessfulReadOperationTimestamp in L50 and L54 be of type Token as well? same for the trimMark in L58, isn't it somehow cleaner to reference any position in the log by the LSN (currently token)? I am doing it in this way in the other PR, just want to be sure there is no argument against this..

Reply:
These refer to different timestamps (i.e system clock time).
Look at how they are being set here:
https://github.com/CorfuDB/CorfuDB/blob/master/generator/src/main/java/org/corfudb/generator/operations/WriteOperation.java#L36

Reply:
Got it!